### PR TITLE
Maestro Prod deployment updates

### PR DESCRIPTION
**Pull request by alexperovich**
Changes:
- Fully implement merge policies
- Fix GitHub PR creation
- Paginate the Subscription History api

### CODE REVIEW

File: src/Maestro/Maestro.MergePolicies/NoExtraCommitsMergePolicy.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+using System.Linq;
```
Comment (jcagme) [inline src/Maestro/Maestro.MergePolicies/NoExtraCommitsMergePolicy.cs]: Nit: might be nice to provide a description of what each policy represents in a comment

File: src/Maestro/SubscriptionActorService/SubscriptionActor.cs [hunk 2]
```diff
@@ -77,6 +79,7 @@ public class SubscriptionActor : ISubscriptionActor, IRemindable
             IReminderManager reminders,
             BuildAssetRegistryContext context,
             IDarcRemoteFactory darcFactory,
+            IEnumerable<Maestro.MergePolicies.MergePolicy> policyEvaluators,
```
Comment (jcagme) [inline src/Maestro/SubscriptionActorService/SubscriptionActor.cs]: MergePolicy only

File: src/Maestro/SubscriptionActorService/SubscriptionActor.cs [hunk 3]
```diff
@@ -85,8 +88,11 @@ public class SubscriptionActor : ISubscriptionActor, IRemindable
             Context = context;
             DarcFactory = darcFactory;
             Logger = logger;
+            PolicyEvaluators = policyEvaluators.ToDictionary(p => p.Name);
         }
 
+        public Dictionary<string, Maestro.MergePolicies.MergePolicy> PolicyEvaluators { get; }
```
Comment (jcagme) [inline src/Maestro/SubscriptionActorService/SubscriptionActor.cs]: MergePolicy only

File: src/Maestro/SubscriptionActorService/SubscriptionActor.cs [hunk 4]
```diff
@@ -291,41 +301,97 @@ private async Task<string> CheckMergePolicyAsyncImpl(string prUrl)
...
+                {
+                    return $"- ✔️ **{policy.DisplayName}** Succeeded";
+                }
+
+                return $"- ❌ **{policy.DisplayName}** {result.Message}";
+            }
+
+            if (results.Any(r => !r.result.Succeeded))
+            {
+
```
Comment (jcagme) [inline src/Maestro/SubscriptionActorService/SubscriptionActor.cs]: Remove blank

File: src/Maestro/cluster-template/parameters/maestro-int.json [hunk 5]
```diff
@@ -11,8 +11,8 @@
     }
   ],
   "certificateSourceKeyVaultName": "maestroint",
-  "clusterCertificate": "maestro-westus2-cloudapp-azure-com",
-  "additionalCertificates": [
+  "certificates": [
+    "maestro-prod-westus2-cloudapp-azure-com"
```
Comment (jcagme) [inline src/Maestro/cluster-template/parameters/maestro-int.json]: `maestro-prod` cert in maestro-int?

File: src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs [hunk 6]
```diff
@@ -196,32 +206,33 @@ private async Task<IEnumerable<DependencyDetail>> GetRequiredUpdatesAsync(string
...
-            List<GitFile> commits = await _gitClient.GetCommitsForPathAsync(repoUri, branch, assetsProducedInCommit, pullRequestBaseBranch);
+            List<GitFile> files = await _gitClient.GetFilesForCommitAsync(repoUri, commit, "eng/common");
 
             _logger.LogInformation($"Generating commits for script files succeeded!");
 
-            return commits;
+            return files;
         }
 
         private async Task CommitFilesForPullRequest(string repoUri, string branch, string assetsProducedInCommit, IEnumerable<DependencyDetail> itemsToUpdate, string pullRequestBaseBranch = null)
```
Comment (jcagme) [inline src/Microsoft.DotNet.Darc/src/DarcLib/Actions/Remote.cs]: Could you please add change this to CommitFilesForPullRequesrAsync?

File: src/Maestro/Maestro.MergePolicies/NoExtraCommitsMergePolicy.cs (hunk 1 above)
Comment (alexperovich) [reply src/Maestro/Maestro.MergePolicies/NoExtraCommitsMergePolicy.cs]: I'll add them.

File: src/Maestro/cluster-template/parameters/maestro-int.json (hunk 5 above)
Comment (alexperovich) [reply src/Maestro/cluster-template/parameters/maestro-int.json]: This cert has a subject alternative name for int.

**Pull request merged**